omod-1.9/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/PatientIdentifierController1_9Test.java

```diff
@@ -114,6 +114,51 @@ public void shouldEditAPatientIdentifier() throws Exception {
 	}
 	
 	@Test
+	public void shouldEditAPatientIdentifierWithPreferredAttibuite() throws Exception {
+		PatientIdentifier patientIdentifier = service.getPatientIdentifierByUuid(getUuid());
+		Boolean originalPreferredStatus = patientIdentifier.getPreferred();
+		Boolean postPreferredStatus = !originalPreferredStatus;
+		SimpleObject postPatientIdentifier = new SimpleObject();
+		postPatientIdentifier.add("preferred", postPreferredStatus);
+		
+		String json = new ObjectMapper().writeValueAsString(postPatientIdentifier);
+		
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/" + getUuid());
+		req.setContent(json.getBytes());
+		handle(req);
+		assertNotEquals(originalPreferredStatus, patientIdentifier.getPreferred());
+	}
+	
+	@Test
+	public void shouldUnsetOtherPreferredIdentifiers() throws Exception {
+		PatientIdentifier nonPreferred = service.getPatientIdentifierByUuid(getUuid());
+		long originalCount = getAllCount();
+		
+		SimpleObject patientIdentifier = new SimpleObject();
+		patientIdentifier.add("identifier", "abc123ez");
+		patientIdentifier.add("identifierType", "2f470aa8-1d73-43b7-81b5-01f0c0dfa53c");
+		patientIdentifier.add("location", RestTestConstants1_8.LOCATION_UUID);
+		patientIdentifier.add("preferred", true);
+		
+		String json = new ObjectMapper().writeValueAsString(patientIdentifier);
+		
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		
+		SimpleObject newPatientIdentifier = deserialize(handle(req));
+		
+		assertNotNull(PropertyUtils.getProperty(newPatientIdentifier, "uuid"));
+		assertEquals(originalCount + 1, getAllCount());
+		
+		String preferredUuid = (String) PropertyUtils.getProperty(newPatientIdentifier, "uuid");
+		PatientIdentifier preferredIdentifer = service.getPatientIdentifierByUuid(preferredUuid);
+		System.out.print("none prefered is " + nonPreferred.isPreferred());
```

Comment:
Have you looked at this? https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Donotprintout